lib/poller.test.js

```diff
@@ -81,13 +84,27 @@ describe('lib.poller.Poller', () => {
       poller.cache = {
         splits: { mockSplit0: { name: 'mockSplit0', status: 'foo' } },
         since: 0,
-        segments: {}
+        segments: {},
+        serializedDataSubsets: {}
       }
       poller.cache.serializedData = 'something-that-should-change'
       await poller.pollForChanges()
       expect(poller.cache.serializedData).to.equal('{"splitsData":{"mockSplit0":"{\\"status\\":\\"bar\\"}","mockSplit1":"{\\"status\\":\\"baz\\"}"},"since":1,"segmentsData":{"mockSegment0":"{\\"name\\":\\"mockSegment0\\",\\"added\\":[]}"},"usingSegmentsCount":1}')
     })
 
+    it('updates the cached serializedDataSubsets', async () => {
+      poller.serializeSegments = true
+      poller.cache = {
+        splits: { mockSplit0: { name: 'mockSplit0', status: 'foo' } },
+        since: 0,
+        segments: {},
+        serializedDataSubsets: { 'mockSplit0.mockSplit1': 'old-cache-data' }
+      }
+      poller.cache.serializedData = 'something-that-should-change'
+      await poller.pollForChanges()
+      expect(poller.cache.serializedDataSubsets).to.deep.equal({ 'mockSplit0.mockSplit1': '{"splitsData":{"mockSplit0":"{\\"status\\":\\"bar\\"}","mockSplit1":"{\\"status\\":\\"baz\\"}"},"since":1,"segmentsData":{"mockSegment0":"{\\"name\\":\\"mockSegment0\\",\\"added\\":[]}"},"usingSegmentsCount":1}' })
```

Comment:
why is there the single `'mockSplit0.mockSplit1'` key instead of one key per experiment?

Reply:
because its the whole script (including segmentsData, since, and usingSegmentsCount) for those two eids